code/game/turfs/simulated/wall_icon.dm

```diff
@@ -177,6 +179,9 @@
 		add_overlay(SSmaterials.wall_damage_overlays[Clamp(round(damage / integrity * DAMAGE_OVERLAY_COUNT) + 1, 1, DAMAGE_OVERLAY_COUNT)])
 
 /turf/simulated/wall/proc/can_join_with(var/turf/simulated/wall/W)
-	if(material && istype(W.material) && get_wall_icon() == W.get_wall_icon())
-		return 1
+	if(material && istype(W.material))
+		if(LAZYACCESS(material.wall_blend_icons, W.get_wall_icon()))
+			return 2
+		if(get_wall_icon() == W.get_wall_icon())
```

Comment:
Since you're checking `W.get_wall_icon()` twice it might be better to declare a var to hold it rather than calling the proc repeatedly.